platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/XMLUtils.java

```diff
@@ -124,28 +135,66 @@ public static Result transform(Node nodeXml, TransformerProperties transformerPr
      */
     public static String getRootNamespace(String xml) {
 
-        String rootNamespace = null;
-
         if (xml == null) {
             return null;
         }
 
-        initializeXMLInputFactory();
+        ProcessingResult<String> result = new ProcessingResult();
+        return processElements(xml, result, (xmlStreamReader) -> {
+            result.setValue(xmlStreamReader.getNamespaceURI());
+            result.done();
+        });
+    }
+
+    /**
+     * This method lets clients parse through the elements of an XML document with less
+     * boiler-plate code. This methods handles the configuration of the
+     * XML factory and disables security risks like macro expansion.
+     * This methods also catches the several exceptions that can be thrown during the creation
+     * and use of an XML parser.
+     * <p>
+     * Example: Get the first three elements whose local name is "Fizz".
+     * <p>
+     * String xml = "my XML document"; // The string representation of the XML document to parse
+     * <p>
+     * XMLUtils.ProcessingResult<List<String>> result =
+     * new XMLUtils.ProcessingResult<>(new ArrayList<String>());
+     * <p>
+     * List<String> output = XMLUtils.processElements(xml, result, (xmlReader) -> {
+     * if (xmlReader.getLocalName().equals("Fizz")) {
+     * if (result.getValue().size() < 3) {
+     * result.getValue().add(xmlReader.getElementText());
```

Comment:
This example concerns me a little as it encourages/forces the mutability of the collection in the result. I don't know that I have a ready solution to that, but it isn't ideal.